### PR TITLE
Setup a nightly build job

### PR DESCRIPTION
**Pull request by award999**
* Renaming workflow file to "test.yml" so we can share the setup, using the event_name to decide if full test matrix should be run or not
* Run insiders nightly

Issue: #934

### CODE REVIEW

**Comment by award999** [comment]
WIP I'm playing around with

**Comment by award999** [comment]
@swift-ci Please test

**Comment by award999** [comment]
@swift-ci Please test

**Comment by award999** [comment]
@swift-ci Please test

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        NODE_VERSION=v18.19.0
+        NODE_PATH=/usr/local/nvm/versions/node/v18.19.0/bin
+        NVM_DIR=/usr/local/nvm
+        CI=1
+      linux_pre_build_command: |
+        apt-get update && apt-get install -y rsync curl gpg libasound2 libgbm1 libgtk-3-0 libnss3 xvfb build-essential
+        mkdir -p $NVM_DIR
+        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+        /bin/bash -c "source $NVM_DIR/nvm.sh && nvm install $NODE_VERSION"
+        echo "$NODE_PATH" >> $GITHUB_PATH
```
Comment (shahmishal) [inline .github/workflows/nightly.yml:23]: Should this be updated to 
```
linux_pre_build_command: . .github/workflows/scripts/setup-linux.sh
```

Comment (award999) [reply .github/workflows/nightly.yml:23]: yep, thanks! Intent was to share but forgot to update

**Review by shahmishal — APPROVED** [review]

**Pull request merged**